src/main/kotlin/app/revanced/patcher/util/patch/PatchBundle.kt

```diff
@@ -2,74 +2,73 @@
 
 package app.revanced.patcher.util.patch
 
-import app.revanced.patcher.data.Context
-import app.revanced.patcher.extensions.PatchExtensions.patchName
 import app.revanced.patcher.patch.Patch
+import app.revanced.patcher.patch.PatchClass
+import dalvik.system.DexClassLoader
 import org.jf.dexlib2.DexFileFactory
 import java.io.File
 import java.net.URLClassLoader
 import java.util.jar.JarFile
 
 /**
  * A patch bundle.
-
- * @param path The path to the patch bundle.
+ *
+ * @param patchClassNames The class names of [Patch]es.
  */
-sealed class PatchBundle(path: String) : File(path) {
-    internal fun loadPatches(classLoader: ClassLoader, classNames: Iterator<String>) = buildList {
-        classNames.forEach { className ->
-            val clazz = classLoader.loadClass(className)
-            if (!clazz.isAnnotationPresent(app.revanced.patcher.patch.annotations.Patch::class.java)) return@forEach
-            @Suppress("UNCHECKED_CAST") this.add(clazz as Class<out Patch<Context>>)
+abstract class PatchBundle(private val patchClassNames: Iterable<String>) {
+
+    internal fun loadPatches(classLoader: ClassLoader) = buildList {
+        patchClassNames.sorted().forEach { className ->
+            classLoader.loadClass(className).let {
+                if (!it.isAnnotationPresent(app.revanced.patcher.patch.annotations.Patch::class.java)) return@forEach
+                @Suppress("UNCHECKED_CAST") this@buildList.add(it as PatchClass)
+            }
         }
```

Comment:
I'm more towards `load` than `get`, because `get` could signal you're just getting something from a container. Meanwhile, in this case, you're not only getting the classes but *also loading them*, which makes the operation much more expensive. Kinda like the `asThis` and `toThis` thingy, you know?

Reply:
In that case `read` should be used conventionally.

Reply:
I have renamed it to `readPatches`